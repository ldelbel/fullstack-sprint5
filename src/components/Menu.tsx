import React, { useContext } from "react";
import CategoriesContext from "../contexts/CategoriesContext";
import { ICategoriesContext } from "../types/interfaces";

interface IMenuItem {
  label: string;
}

function MenuItem({ label }: IMenuItem) {
  return (
    <li className="menu__item">
      <a className="menu__link" href="#home">
        <span>{label}</span>
      </a>
    </li>
  );
}

function Menu() {
  const { categories } = useContext<ICategoriesContext>(CategoriesContext);

  return (
    <nav className="header__menu menu">
      <ul className="menu__list">
        {categories.all &&
          categories.all.map((m) => <MenuItem key={m.id} label={m.label} />)}
      </ul>
    </nav>
  );
}

export default Menu;
