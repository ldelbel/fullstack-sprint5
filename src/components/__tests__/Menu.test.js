import React from "react";
import { render } from "@testing-library/react";
import renderer from 'react-test-renderer';
import Menu from "../Menu";

jest.mock("react", () => {
  const react = jest.requireActual("react");
  return {
    ...react,
    useContext: () => ({
      categories: {
        all: [
          {
            id: 1,
            label: "Novidades",
            link: "#home",
          },
          {
            id: 2,
            label: "Feminino",
            link: "#home",
          },
        ],
      },
    }),
  };
});

describe("Menu component", () => {
  it("should render categories", () => {
    const { getByText } = render(<Menu />);
    expect(getByText("Novidades")).toBeInTheDocument();
    expect(getByText("Feminino")).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const menu = renderer.create(<Menu />).toJSON();
    expect(menu).toMatchSnapshot();
  })
});
