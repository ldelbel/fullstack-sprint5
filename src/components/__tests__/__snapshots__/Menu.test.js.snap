// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component should match snapshot 1`] = `
<nav
  className="header__menu menu"
>
  <ul
    className="menu__list"
  >
    <li
      className="menu__item"
    >
      <a
        className="menu__link"
        href="#home"
      >
        <span>
          Novidades
        </span>
      </a>
    </li>
    <li
      className="menu__item"
    >
      <a
        className="menu__link"
        href="#home"
      >
        <span>
          Feminino
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
